pkg/crypto/keys/publickey.go

```diff
@@ -34,6 +35,34 @@ func (keys PublicKeys) Less(i, j int) bool {
 	return keys[i].Y.Cmp(keys[j].Y) == -1
 }
 
+// DecodeBytes decodes a PublicKeys from the given slice of bytes.
+func (keys PublicKeys) DecodeBytes(data []byte) error {
+	b := io.NewBinReaderFromBuf(data)
+	b.ReadArray(keys)
+	return b.Err
+}
+
+// Contains checks whether passed param contained in PublicKeys.
+func (keys PublicKeys) Contains(pKey *PublicKey) bool {
+	for _, key := range keys {
+		if reflect.DeepEqual(key, pKey) {
```

Comment:
Do you really need to use reflection here? I think it's easier to introduce proper `Equals` method for `PublicKey` that will compare `X` and `Y` and then use it here.